textproc/pandoc/Portfile

```diff
@@ -24,33 +25,45 @@ long_description    \
     Textile, groff man pages, plain text, Emacs Org-Mode, AsciiDoc, EPUB (v2 \
     and v3), FictionBook2, and S5, Slidy and Slideous HTML slide shows.
 
-depends_lib-append  \
-                    port:hs-aeson \
-                    port:hs-alex \
-                    port:hs-attoparsec \
-                    port:hs-base64-bytestring \
-                    port:hs-blaze-html \
-                    port:hs-blaze-markup \
-                    port:hs-data-default \
-                    port:hs-extensible-exceptions \
-                    port:hs-happy \
-                    port:hs-highlighting-kate \
-                    port:hs-hslua \
-                    port:hs-http \
-                    port:hs-mtl \
-                    port:hs-network \
-                    port:hs-pandoc-types \
-                    port:hs-parsec \
-                    port:hs-random \
-                    port:hs-syb \
-                    port:hs-sha \
-                    port:hs-tagsoup \
-                    port:hs-temporary \
-                    port:hs-texmath \
-                    port:hs-text \
-                    port:hs-unordered-containers \
-                    port:hs-vector \
-                    port:hs-xml \
-                    port:hs-yaml \
-                    port:hs-zip-archive \
-                    port:hs-zlib
+depends_build-append  \
+                    port:stack
+
+set system_gcc      /usr/bin/gcc
```

Comment:
Why is this needed? Why cannot other compilers be used?

Reply:
Indeed. Please do nit hardcode a specific compiler like this. Any reason why this cannot just be ${configure.cc} ?

Reply:
Stack builds work using native `gcc`, but break using out-of-the-box MacPorts gcc.

I’m sure there’s a fix, but I never had time to track it down so just punted and used native `gcc` until someone can figure out the fix.

Reply:
We don't want to build anything with the "real" GNU `gcc`, and in case the build was calling gnu gcc from MacPorts, that was a flaw that shouldn't have happened.

What you call "native `gcc`" (`/usr/bin/gcc`) is in fact clang on any modern macOS.

Can you please try using `${configure.compiler}` at the place where you used `/usr/bin/gcc` (or perhaps replace `system_gcc` with `configure.compiler` and maybe don't try to change the `PATH`)? In what way does that fail? (You should leave all explicit settings of the compiler as you have them now, just use `${configure.compiler}` instead of `/usr/bin/gcc`.)

Reply:
Completely agree with @mojca. Please use `${configure.compiler}` and if there are issues they should be addressed. hardcoding `/usr/bin/gcc` is really not an acceptable solution.

Reply:
Okay, I replaced with `${configure.compiler}`.

If this builds, I'll single-step through the other comments and make sure each delta produces a successful build.

I'm also wondering about the necessity of those `.env` statements too …

Reply:
`stack` needs the whole path to the compiler executable, not simply "`clang`".

How is this done within a Portfile?

See: https://paste.z0k.xyz/8b7e952d4573
```
DEBUG: system -W /opt/local/var/macports/build/_Users_vsts_agent_2.155.1_work_1_s_textproc_pandoc/pandoc/work/pandoc-2.7.3: /opt/local/bin/stack setup --stack-root /opt/local/var/macports/build/_Users_vsts_agent_2.155.1_work_1_s_textproc_pandoc/pandoc/work/.stack  --with-gcc clang --allow-different-user  && /opt/local/bin/stack build --stack-root /opt/local/var/macports/build/_Users_vsts_agent_2.155.1_work_1_s_textproc_pandoc/pandoc/work/.stack  --with-gcc clang --allow-different-user  && /opt/local/bin/stack install --stack-root /opt/local/var/macports/build/_Users_vsts_agent_2.155.1_work_1_s_textproc_pandoc/pandoc/work/.stack  --local-bin-path /opt/local/var/macports/build/_Users_vsts_agent_2.155.1_work_1_s_textproc_pandoc/pandoc/work/pandoc-2.7.3/bin  --with-gcc clang --allow-different-user
option --with-gcc: InvalidAbsFile "clang"
```

Reply:
try with `${configure.cc}` that gives a complete path.

Reply:
Please excuse me, @cjones051073 is right, it should be `configure.cc` instead. I would be surprised if that doesn't set the full path to the compiler.

Reply:
It does.

Reply:
That worked. Next I’ll implement the test stuff so it catches any issues when I implement the rest of the changes.

Reply:
> BTW. The normal way we handle situations like that is to use `compiler.blacklist`.

Sure, but only once you have [TheRightCompiler](https://trac.macports.org/wiki/UsingTheRightCompiler) working.